main/views.py

```diff
@@ -235,3 +260,162 @@ def delete_profile(request):
+        )
+
+        payroll.calculate_final_salary()
+
+        return JsonResponse({"status": 201, "data": payroll.to_dict()})
+
+    except Profile.DoesNotExist:
+        return JsonResponse({"status": 404, "message": "Profile not found for the logged-in user"})
+    except Exception as e:
+        print(f"Error in create_payroll: {e}")
+        return JsonResponse({"status": 500, "message": "Internal Server Error"})
+
+
+@csrf_exempt
+@auth_required
+def generate_employee_pdf(request):
+    """
+    Generate a PDF for the logged-in user with their Profile and Payroll details.
+    """
+    try:
+        user = request.user 
+        pdf_directory = os.path.join(settings.BASE_DIR, "pdfs")
+        os.makedirs(pdf_directory, exist_ok=True)
+
+        profile = Profile.objects.filter(user=user).first()
+        if not profile:
+            return HttpResponse("Profile data not found for the logged-in user.", status=404)
+
+        payroll = Payroll.objects.filter(profile=profile).first()
+        if not payroll:
+            return HttpResponse("Payroll data not found for the logged-in user.", status=404)
+
+        pdf_path = os.path.join(pdf_directory, f"{user.username}_details.pdf")
+        pdf = canvas.Canvas(pdf_path, pagesize=letter)
+        pdf.setTitle("Employee Payroll Details")
+        pdf.setFont("Helvetica-Bold", 16)
+        pdf.drawString(200, 750, "Employee Payroll Details")
+        pdf.setFont("Helvetica", 12)
+
+        y_position = 700
+        pdf.drawString(50, y_position, f"Name: {user.get_full_name()}")
+        y_position -= 20
+        pdf.drawString(50, y_position, f"Username: {user.username}")
+        y_position -= 20
+        pdf.drawString(50, y_position, f"Phone: {profile.phone or 'N/A'}")
+        y_position -= 20
+        pdf.drawString(50, y_position, f"Address: {profile.address or 'N/A'}")
+        y_position -= 20
+        pdf.drawString(50, y_position, f"Job Title: {profile.job_title or 'N/A'}")
+        y_position -= 20
+        pdf.drawString(50, y_position, f"Date of Joining: {profile.date_of_joining or 'N/A'}")
+        y_position -= 20
+
+        pdf.setFont("Helvetica-Bold", 14)
+        y_position -= 20
+        pdf.drawString(50, y_position, "Payroll Details")
+        pdf.setFont("Helvetica", 12)
+        y_position -= 20
+        pdf.drawString(50, y_position, f"Basic Salary: {payroll.basic_salary}")
+        y_position -= 20
+        pdf.drawString(50, y_position, f"HRA: {payroll.hra}")
+        y_position -= 20
+        pdf.drawString(50, y_position, f"Special Allowance: {payroll.special_allowance}")
+        y_position -= 20
+        pdf.drawString(50, y_position, f"Tax Deduction: {payroll.tax_deduction}")
+        y_position -= 20
+        pdf.drawString(50, y_position, f"Reimbursement Amount: {payroll.reimbursement_amount}")
+        y_position -= 20
+        pdf.drawString(50, y_position, f"Final Salary: {payroll.final_salary}")
```

Comment:
ok,  so the final salary in LPA is directly send to the employee as it's monthly salary. Great !